api/main.py

```diff
@@ -138,6 +139,42 @@ async def post_user(
     return obj
 
 
+@app.post('/group', response_model=UserGroup, response_model_by_alias=False)
+async def post_user_group(
+        group: UserGroup,
+        current_user: User = Security(get_user, scopes=["admin"])):
+    """Create new user group"""
+    try:
+        obj = await db.create(group)
+    except DuplicateKeyError as error:
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            detail=f"Group '{group.name}' already exists. \
+Use a different group name."
+        ) from error
+    await pubsub.publish_cloudevent('user_group', {'op': 'created',
+                                                   'id': str(obj.id)})
+    return obj
+
+
+@app.get('/groups', response_model=PageModel)
+async def get_user_groups(request: Request):
+    """Get all the user groups if no request parameters have passed.
+       Get all the matching user groups otherwise."""
```

Comment:
By the way, I thought docstrings were meant to have only a one-line summary first and then a blank line and paragraphs like git commits:
https://peps.python.org/pep-0257/#multi-line-docstrings

Following this convention, it could be reworded with something like "Find user groups" as a one-line summary and then the full description as a paragraph.

Reply:
Ah but I see this convention wasn't followed for the other endpoints either, so that's not really an issue with this PR.  I guess we can just have a follow-up to enable `pydocstyle` and fix all the warnings.